ui-admin/src/components/Tree.tsx

```diff
@@ -0,0 +1,256 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faChevronDown, faChevronRight } from '@fortawesome/free-solid-svg-icons'
+import { get } from 'lodash'
+import React, { Dispatch, SetStateAction, useRef, useState } from 'react'
+
+export type TreeItemT<T> = {
+  label: string
+  data: T
+  children?: TreeItemT<T>[]
+}
+
+type TreeItemProps<T> = {
+  activeDescendant: string
+  id: string
+  isItemSelected: (item: TreeItemT<T>) => boolean
+  item: TreeItemT<T>
+  level: number
+  path: string
```

Comment:
I didn't see anywhere where you documented what the path string format is (which isn't a big deal since it's internal to this component.)  But It looks like it's basic object notation?
```suggestion
  path: string // object notation path string e.g. 'pages[1].elements[3]`
```